### ISSUE TITLE
oneOf with discriminator

### ISSUE THREAD

**Issue by inomn** [acacode/swagger-typescript-api#558]
Hi I have an issue with `oneOf` with `discriminator` generation

Schema:
```
"MessageAuthor": {
        "title": "MessageAuthor",
        "oneOf": [
          {
            "$ref": "#/components/schemas/UserAuthor"
          },
          {
            "$ref": "#/components/schemas/OperatorAuthor"
          },
        ],
        "discriminator": {
          "propertyName": "type",
          "mapping": {
            "USER": "#/components/schemas/UserAuthor",
            "OPERATOR": "#/components/schemas/OperatorAuthor",
          }
        }
```

Before (v11):

```
export type MessageAuthor = UserAuthor | OperatorAuthor;
```

After (v13):
```
export type MessageAuthor = BaseMessageAuthor &
  (
    | BaseMessageAuthorTypeMapping<"USER", UserAuthor>
    | BaseMessageAuthorTypeMapping<"OPERATOR", OperatorAuthor>
  );

type BaseMessageAuthor = any;

type BaseMessageAuthorTypeMapping<Key, Type> = {
  type: Key;
} & Type;
```

Also, https://github.com/acacode/swagger-typescript-api/issues/459 is fixed in `13.0.0`, but still the case in `13.0.2`.

**Comment by inomn** [comment]
To add more details: i'm concerned about "Base" types, typed as "any".
It kinda of breaks the app and adds a lot of overhead.

Base type should \ could contain all common fields, but instead it's "any".

**Comment by js2me** [comment]
@inomn hello, yes I know about this problem. I think best solution of this is to remove "any" type from generated data contract and result type will be

export type MessageAuthor =
  (
    | BaseMessageAuthorTypeMapping<"USER", UserAuthor>
    | BaseMessageAuthorTypeMapping<"OPERATOR", OperatorAuthor>
  );

**Comment by js2me** [comment]
Or replace any to Record<string, any>

**Comment by js2me** [comment]
@inomn I think this type is not correct for open api discriminator with mapping declaration

export type MessageAuthor = UserAuthor | OperatorAuthor;

**Comment by inomn** [comment]
@js2me hi, thanks for the response. 

```
export type MessageAuthor = UserAuthor | OperatorAuthor;
```

is how swagger-typescript-api v11 generated the same scheme. I agree it's not really correct, I provided it as an example.\

>  I think best solution of this is to remove "any" type from generated data contract and result type will be ...
You mean manually after type generation? Or it's something you want to implement in next versions?

**Comment by inomn** [comment]
Thanks! Fixed in 13.0.3
